app/components/post-search-results-item.js

```diff
@@ -0,0 +1,21 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+
+export default Component.extend({
+  isQuizWithTakes: computed('post.{type,stats,stats.takes}', function () {
+    const type = this.post.type;
+    const stats = this.post.stats;
+
+    return type === 'quiz' && stats && stats.takes && parseInt(stats.takes, 10) > 0;
+  }),
+
+  imageThumbnail: computed('post.image', function () {
+    const image = this.post.image;
+
+    return window.Vignette ? window.Vignette.getThumbURL(image, {
+      mode: window.Vignette.mode.zoomCrop,
```

Comment:
Do you have to check for these? or can we safely assume those props will be there?

Reply:
It's copied over from another component.

Reply:
I don't think the check is needed, the vignette-js lib is loaded here https://github.com/Wikia/mobile-wiki/blob/8ddea9e2f7cca5f40bd969c23ec092b241d73c4b/ember-cli-build.js#L147, there are actual several places where it is used without check. Also, instead of using `window.Vignette` you could use 
`import Thumbnailer from '../modules/thumbnailer';`
and then `Thumbnailer.getThumbURL(...)`

Reply:
Fixed